virtcontainers/container.go

```diff
@@ -545,6 +545,15 @@ func (c *Container) shareFiles(m Mount, idx int, hostSharedDir, guestSharedDir s
 func (c *Container) mountSharedDirMounts(hostSharedDir, guestSharedDir string) (map[string]Mount, map[string]Mount, error) {
 	sharedDirMounts := make(map[string]Mount)
 	ignoredMounts := make(map[string]Mount)
+	var err error
+	var devicesToDetach []string
+	defer func() {
+		if err != nil {
+			for _, id := range devicesToDetach {
+				c.sandbox.devManager.DetachDevice(id, c.sandbox)
```

Comment:
If I understand correctly, if the mounting fails, today we return an error, and the container creation flow fails.  Shouldn't the VM be getting torn down already?

In that scenario, are we failing to undo the bind mount on the host?

@amshinde can you help here?

Reply:
Ping @amshinde.

Reply:
We are hotplugging block devices to the VM here. If the `storeSandboxDevices` function fails, then the container will fail, but the VM may not be brought down with other containers running. So, this seems like valid use case, although if we fail to write to file system, we may be running into much bigger issues, eventually causing the pod/VM to fail.

Reply:
@egernst 
Do you have any more comment ?